flink-core/src/main/java/org/apache/flink/configuration/AkkaOptions.java

```diff
@@ -39,14 +42,24 @@
                                     + "memory footprint.");
 
     /** Timeout for akka ask calls. */
-    public static final ConfigOption<String> ASK_TIMEOUT =
+    public static final ConfigOption<Duration> ASK_TIMEOUT_DURATION =
             ConfigOptions.key("akka.ask.timeout")
-                    .stringType()
-                    .defaultValue("10 s")
+                    .durationType()
+                    .defaultValue(Duration.ofSeconds(10))
                     .withDescription(
                             "Timeout used for all futures and blocking Akka calls. If Flink fails due to timeouts then you"
                                     + " should try to increase this value. Timeouts can be caused by slow machines or a congested network. The"
                                     + " timeout value requires a time-unit specifier (ms/s/min/h/d).");
```

Comment:
Is the mention of requiring a time-unit specifier still relevant? I feel this was important when it was string type, but it's automatically included if the type is `Duration`
(changing it would mean that the deprecated `ASK_TIMEOUT` should have a different description, though)

Reply:
I think it is still required because the way we parse durations is still custom.

Reply:
But isn't the "timeout value" that is referenced in the description the value associated with this config key? Since its type is `Duration`, I don't see where the time-unit specifier comes into play now.
I also just checked the `toString` implementation of `Duration` and that looks completely different...

Reply:
It comes into play when configuring this option via the flink-conf.yaml, where you'd specify something like:
```
akka.ask.timeout: 10s
```
This is then parsed via `org.apache.flink.util.TimeUtils#parseDuration`.

Reply:
I see, thanks!